routing/pathfind.go

```diff
@@ -502,29 +507,41 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		defer tx.Rollback()
 	}
 
-	if len(r.DestCustomRecords) > 0 {
-		// Check if the target has TLV enabled
-
+	// If no destination features are provided, we will load what features
+	// we have for the target node from our graph.
+	features := r.DestFeatures
+	if features == nil {
```

Comment:
i think it makes sense to keep this one in particular, otherwise we will have no fallback option. how can we distinguish between an (intentionally) empty feature vector and not having one at all? similar to my comment below, this check happens before starting pathfinding, so we haven't yet used `finalHopParams`